### PR TITLE
docs(connectors): add result mapping doc

### PR DESCRIPTION
**Pull request by igpetrov**
## What is the purpose of the change

docs(connectors): add result mapping doc.

Related to https://github.com/camunda/connectors-bundle/issues/158.

## Are there related marketing activities

No.

## When should this change go live?

Anytime. This is an enhancement / clarification to existing doc.

## PR Checklist

- [x] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by igpetrov** [comment]
@bastiankoerber, please, take a look

File: docs/components/connectors/use-connectors.md [hunk 1]
```diff
@@ -97,6 +97,70 @@ For further details on how secrets are implemented in Connectors, consult our [C
...
+    }
+  }
+}
+```
+
+If you declare a variable `myWeatherResponse` in the `Result Variable` field, the whole response will be mapped to the declared variable.
+
+Now, let's imagine that you wish to extract only temperature into a process variable `berlinWeather` and wind speed into `berlinWindSpeed`. Let's also imagine, you need weather in Fahrenheit declared in `berlinWeatherInFahrenheit`.
+
+In that case, you could declare `Result Expression` as follows:
```
Comment (bastiankoerber) [inline docs/components/connectors/use-connectors.md:150]: maybe we can add a sentence that only the values defined in the result expression are stored in the result variable "myWeatherResponse". Just to make it explicit.

Comment (bastiankoerber) [reply docs/components/connectors/use-connectors.md:150]: maybe as last sentence under the response mapping result image

**Review by bastiankoerber — COMMENTED** [review]
i added one suggestion. Rest looks pretty clear and straight forward.

After you merged it, can you add a note in our guideline that we are link this page in our property description for result expression?

**Review by christinaausley — APPROVED** [review]
Technical adjustments in latest commit, including capitalizing Connector and incorporating an admonition. Let me know if you have questions about these. LGTM!

**Review by christinaausley — APPROVED** [review]

File: docs/components/connectors/use-connectors.md (hunk 1 above)
Comment (igpetrov) [reply docs/components/connectors/use-connectors.md:152]: Sorry, didn't get your comment. The variable `myWeatherResponse` will store everything and it's not related to result expression.

**Pull request merged**